python/ray/dataframe/dataframe.py

```diff
@@ -3260,7 +3332,7 @@ def _getitem_array(self, key):
             new_parts = _map_partitions(lambda df: df[key],
                                         self._col_partitions)
             columns = self.columns
-            index = self.index[key]
+            index = self._col_metadata[key].index
```

Comment:
Why did this change? This should not use `_col_metadata`, but instead `_row_metadata`.